src/utils.jl

```diff
@@ -46,134 +40,133 @@ end
 """
-sinhcu(x) = _sinhcu(float(x))
-function _sinhcu(x::Union{T,Complex{T}}) where {T}
-    taylor_0_bound = eps(T)
-    taylor_2_bound = sqrt(taylor_0_bound)
-    taylor_n_bound = sqrt(taylor_2_bound)
-
-    if isinf_imag(x) 
-        return zero(x)
-    end
-    
-    nrm = Base.Math.fastabs(x)
+function sinhcu(x)
+  #if isinf_imag(x)
+  #  return zero(x)
+  #end
 
-    if nrm >= taylor_n_bound || isnan(nrm)
-        return sinh(x)/x
-    else
-        # approximation by taylor series in x at 0 up to order 0
-        res = one(x)
-        if nrm >= taylor_0_bound
-            x2 = x*x
-            # approximation by taylor series in x at 0 up to order 2
-            res += x2/6
-            if nrm >= taylor_2_bound
-                # approximation by taylor series in x at 0 up to order 4
-                res += (x2*x2)/120
-            end
-        end
-        return res
-    end
+  threshold = 0.0004 # (120*eps(Float64))^(1/4)
+  return vifelse(abs(x) > threshold, sinh(x)/x, 1+x*x/6)
+end
+
+
+function atan2(y, x)
+  return atan(y, x)
+end
+
+
+function atan2(y::Vec{N, T}, x::Vec{N, T}) where {N, T}
+  arctan = atan(y/x)
+  return vifelse(x > 0, arctan,
+         vifelse((x < 0)  & (y >= 0),  arctan + Vec{N, T}(T(pi)),
+         vifelse((x < 0)  & (y < 0),   arctan - Vec{N, T}(T(pi)),
+         vifelse((x == 0) & (y > 0),   Vec{N, T}(pi/2),
+         vifelse((x == 0) & (y < 0),   Vec{N, T}(-pi/2),
+         vifelse((x == 0) & (y == 0),  Vec{N, T}(0), 
+         Vec{N, T}(NaN)))))))
 end
 
-# Copied pasted from sincc in bmad-ecosystem
+
+# Copy-pasted from sincc in bmad-ecosystem
 function sincuc(x) 
-  if Base.Math.fastabs(x) < 0.1
-    c0 = 1/6
-    c1 = -1/120
-    c2 = 1/5040
-    c3 = -1/362880
-    x2 = x^2
-    y = c0 + x2 * (c1 + x2 * (c2 + x2 * c3))
-  else
-    y = (x - sin(x)) / x^3
-  end
-  return y
+  c0 = 1/6
+  c1 = -1/120
+  c2 = 1/5040
+  c3 = -1/362880
+  x2 = x^2
+  return vifelse(abs(x) >= 0.1, (x-sin(x))/x^3, c0+x2*(c1+x2*(c2+x2*c3)))
 end
 
-"""
-    sincus(x)
 
-Compute the unnormalized sinc square-root function 
-``\\operatorname{sincus}(x) = \\sin(\\sqrt(x)) / (\\sqrt(x))`` 
-with accuracy near the origin.
-"""
-sincus(x) = _sincus(float(x))
-function _sincus(x::Union{T,Complex{T}}) where {T}
-    nrm = Base.Math.fastabs(x)
-    if nrm >= 109.018*eps(T)^(1/11)
-        return sin(sqrt(x))/(sqrt(x))
-    else
-        c0 = 1
-        c1 = -1/6
-        c2 = 1/120
-        c3 = -1/5040
-        c4 = 1/362880
-        c5 = -1/39916800
-        c6 = 1/6227020800
-        c7 = -1/1307674368000
-        c8 = 1/355687428096000
-        c9 = -1/12164510040883200000
-        c10 = 1/5109094217170944000000
-        return (c0+x*(c1+x*(c2+x*(c3+x*(c4+x*(c5+x*(c6+x*(c7+x*(c8+x*
-        (c9+x*c10))))))))))
-    end
+function sincos_quaternion(x)
+  """
+  This function computes sin(sqrt(x))/sqrt(x) and cos(sqrt(x)), which are both 
+  necessary for exponentiating a rotation vector into a quaternion.
+  """
```

Comment:
```suggestion
"""
  This function computes sin(sqrt(x))/sqrt(x) and cos(sqrt(x)), which are both 
  necessary for exponentiating a rotation vector into a quaternion.
"""
function sincos_quaternion(x)
```

Comments should be above the function

Reply:
Can you fix that for each of these functions?